core/src/main/java/com/linecorp/armeria/common/SessionProtocol.java

```diff
@@ -17,93 +17,53 @@
 
-import com.linecorp.armeria.common.http.HttpSessionProtocols;
-
 /**
  * Session-level protocol that provides facilities such as framing and flow control.
  */
-public final class SessionProtocol implements Comparable<SessionProtocol> {
-
-    private static final BiMap<String, SessionProtocol> uriTextToProtocols;
-    private static final Set<SessionProtocol> values;
-
-    static {
-        // Load all session protocols from the providers.
-        final BiMap<String, SessionProtocol> mutableUriTextToProtocols = HashBiMap.create();
-        ServiceLoader.load(SessionProtocolProvider.class, SessionProtocolProvider.class.getClassLoader())
-                     .forEach(provider -> provider.entries().forEach(e -> {
-                         checkState(!mutableUriTextToProtocols.containsKey(e.uriText),
-                                    "session protocol registered already: ", e.uriText);
-
-                         mutableUriTextToProtocols.put(e.uriText,
-                                 new SessionProtocol(e.uriText, e.useTls, e.isMultiplex, e.defaultPort));
-                     }));
-
-        uriTextToProtocols = ImmutableBiMap.copyOf(mutableUriTextToProtocols);
-        values = uriTextToProtocols.values();
-    }
-
+public enum SessionProtocol {
     /**
-     * @deprecated Use {@link HttpSessionProtocols#HTTPS} instead.
+     * HTTP - over TLS, HTTP/2 preferred.
      */
-    @Deprecated
-    public static final SessionProtocol HTTPS = HttpSessionProtocols.HTTPS;
-
+    HTTPS("https", true, false, 443),
     /**
-     * @deprecated Use {@link HttpSessionProtocols#HTTP}.
+     * HTTP - cleartext, HTTP/2 preferred.
      */
-    @Deprecated
-    public static final SessionProtocol HTTP = HttpSessionProtocols.HTTP;
-
+    HTTP("http", false, false, 80),
     /**
-     * @deprecated Use {@link HttpSessionProtocols#H1}.
+     * HTTP/1 - over TLS.
      */
-    @Deprecated
-    public static final SessionProtocol H1 = HttpSessionProtocols.H1;
-
+    H1("h1", true, false, 443),
     /**
-     * @deprecated Use {@link HttpSessionProtocols#H1C}.
+     * HTTP/1 - cleartext.
      */
-    @Deprecated
-    public static final SessionProtocol H1C = HttpSessionProtocols.H1C;
-
+    H1C("h1c", false, false, 80),
     /**
-     * @deprecated Use {@link HttpSessionProtocols#H2}.
+     * HTTP/2 - over TLS.
      */
-    @Deprecated
-    public static final SessionProtocol H2 = HttpSessionProtocols.H2;
-
+    H2("h2", true, true, 443),
     /**
-     * @deprecated Use {@link HttpSessionProtocols#H2C}.
+     * HTTP/2 - cleartext.
      */
-    @Deprecated
-    public static final SessionProtocol H2C = HttpSessionProtocols.H2C;
+    H2C("h2c", false, true, 80);
 
-    /**
-     * @deprecated Use {@link HttpSessionProtocols#values()}.
-     */
-    @Deprecated
-    public static Set<SessionProtocol> ofHttp() {
-        return HttpSessionProtocols.values();
-    }
+    private static final BiMap<String, SessionProtocol> uriTextToProtocols;
```

Comment:
I don't see any usages of the reversed map - do you have future plans for using this `BiMap`?